ansible_collections/arista/avd/roles/eos_designs/python_modules/metadata/cv_pathfinder.py

```diff
@@ -0,0 +1,122 @@
+                "regions": self._metadata_regions(),
+                "vrfs": self._metadata_vrfs(),
+            }
+
+        # Edge or transit
+        return {
+            "role": self.shared_utils.cv_pathfinder_role,
+            "vtep_ip": self.shared_utils.router_id,
+            "region": self.shared_utils.wan_region["name"],
+            "zone": self.shared_utils.wan_zone["name"],
+            "site": self.shared_utils.wan_site["name"],
+            "interfaces": self._metadata_interfaces(),
+            "pathfinders": self._metadata_pathfinder_vtep_ips(),
+        }
+
+    def _metadata_interfaces(self: AvdStructuredConfigMetadata) -> list:
+        return [
+            {
+                "name": interface["name"],
+                "carrier": carrier["name"],
+                "circuit_id": interface.get("wan_circuit_id"),
+                "pathgroup": carrier["path_group"],
+                "public_ip": str(interface["ip_address"]).split("/", maxsplit=1)[0] if self.shared_utils.cv_pathfinder_role == "pathfinder" else None,
+            }
+            for carrier in self.shared_utils.wan_local_carriers
+            for interface in carrier["interfaces"]
+        ]
+
+    def _metadata_pathgroups(self: AvdStructuredConfigMetadata) -> list:
+        return [
+            {
+                "name": pathgroup["name"],
+                "carriers": [
+                    {
+                        "name": carrier["name"],
+                    }
+                    for carrier in self.shared_utils.wan_carriers
+                    if carrier["path_group"] == pathgroup["name"]
+                ],
+                "imported_carriers": [
+                    {
+                        "name": carrier["name"],
+                    }
+                    for carrier in self.shared_utils.wan_carriers
+                    if carrier["path_group"] in [imported_pathgroup["remote"] for imported_pathgroup in pathgroup.get("import_path_groups", [])]
+                ],
+            }
+            for pathgroup in self.shared_utils.wan_path_groups
+        ]
+
+    def _metadata_regions(self: AvdStructuredConfigMetadata) -> list:
+        regions = get(
+            self._hostvars, "cv_pathfinder_regions", required=True, org_key="'cv_pathfinder_regions' key must be set when 'wan_mode' is 'cv-pathfinder'."
+        )
+        return [
+            {
+                "name": region["name"],
+                "id": region["id"],
+                "zones": [
+                    {
+                        # TODO: Once we give configurable zones this should be updated
+                        "name": self.shared_utils.wan_zone["name"],
+                        "id": self.shared_utils.wan_zone["id"],
+                        "sites": [
+                            {
+                                "name": site["name"],
+                                "id": site["id"],
+                                "location": site.get("location"),
```

Comment:
this may need to be changed once we try integration given that `address` is expected in the schema inside location

Reply:
I will fix it.